test/InputRangeUtilsTest.cpp

```diff
@@ -80,6 +80,25 @@ TEST(CachingTransformInputRange, BasicTests) {
                                                             firstPlusTwo);
 }
 
+// Test for iterating past the end of a CachingTransformInput range
```

Comment:
Also add tests for the new abstraction (the concatenatiion)
 + the behavior that you can call " begin"  and " get" and whatnotelse multiple times on any `InputRangeFromGet` thing.